src/routers/StackRouter.js

```diff
@@ -110,15 +110,25 @@ export default (
           action.type === NavigationActions.NAVIGATE &&
           childRouters[action.routeName] !== undefined
         ) {
+          const childRouter = childRouters[action.routeName];
+          let childState;
+          if (childRouter !== null) {
+            const childAction =
+              action.action ||
```

Comment:
I don't think the first action should leak into the initialization logic at all, so I don't think we should pass `action.action` to the child here. Instead, let the following logic handle it (see my next comment)